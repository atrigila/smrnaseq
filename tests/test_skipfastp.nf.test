
nextflow_pipeline {

    name "Test Workflow main.nf - test_skipfastp"
    script "main.nf"
    profile "test_skipfastp"
    tag "test_skipfastp"
    tag "pipeline"

    test("test_skipfastp") {

        when {
            params {
                outdir = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(UTILS.removeNextflowVersion("$outputDir")).match("software_versions") },
                { assert workflow.trace.succeeded().size() == 41 },

                { assert snapshot(
                    path("$outputDir/mirna_quant/reference/hairpin.fa_igenome.fa"),
                    path("$outputDir/mirna_quant/reference/mature.fa_igenome.fa"),
                    path("$outputDir/mirna_quant/reference/hairpin.fa_igenome.fa_idx.fa"),
                    path("$outputDir/mirna_quant/reference/mature.fa_igenome.fa_idx.fa")
                ).match("mirna_quant_reference") },

                { assert snapshot(
                    path("$outputDir/mirna_quant/seqcluster/final/small_Clone1_N3.fastp_trimmed.fastq.gz"),
                    path("$outputDir/mirna_quant/seqcluster/final/small_Clone1_N1.fastp_trimmed.fastq.gz")
                ).match("mirna_quant_seqcluster") },

                { assert snapshot(
                    path("$outputDir/mirna_quant/bam/mature/Clone1_N3_mature.sorted.bam").exists(),
                    path("$outputDir/mirna_quant/bam/mature/Clone1_N1_mature.sorted.bam").exists(),
                    path("$outputDir/mirna_quant/bam/mature/Clone1_N3_mature.sorted.bam.bai").exists(),
                    path("$outputDir/mirna_quant/bam/mature/Clone1_N1_mature.sorted.bam.bai").exists(),
                    path("$outputDir/mirna_quant/bam/mature/Clone1_N3_mature.sorted.stats"),
                    path("$outputDir/mirna_quant/bam/mature/Clone1_N1_mature.sorted.stats"),
                    path("$outputDir/mirna_quant/bam/mature/Clone1_N1_mature.sorted.idxstats"),
                    path("$outputDir/mirna_quant/bam/mature/Clone1_N3_mature.sorted.idxstats"),
                    path("$outputDir/mirna_quant/bam/mature/Clone1_N3_mature.sorted.flagstat"),
                    path("$outputDir/mirna_quant/bam/mature/Clone1_N1_mature.sorted.flagstat"),
                    path("$outputDir/mirna_quant/bam/hairpin/Clone1_N3_mature_hairpin.sorted.bam").exists(),
                    path("$outputDir/mirna_quant/bam/hairpin/Clone1_N1_mature_hairpin.sorted.bam").exists(),
                    path("$outputDir/mirna_quant/bam/hairpin/Clone1_N3_mature_hairpin.sorted.bam.bai").exists(),
                    path("$outputDir/mirna_quant/bam/hairpin/Clone1_N1_mature_hairpin.sorted.bam.bai").exists(),
                    path("$outputDir/mirna_quant/bam/hairpin/Clone1_N1_mature_hairpin.sorted.flagstat"),
                    path("$outputDir/mirna_quant/bam/hairpin/Clone1_N3_mature_hairpin.sorted.stats"),
                    path("$outputDir/mirna_quant/bam/hairpin/Clone1_N1_mature_hairpin.sorted.stats"),
                    path("$outputDir/mirna_quant/bam/hairpin/Clone1_N1_mature_hairpin.sorted.idxstats"),
                    path("$outputDir/mirna_quant/bam/hairpin/Clone1_N3_mature_hairpin.sorted.flagstat"),
                    path("$outputDir/mirna_quant/bam/hairpin/Clone1_N3_mature_hairpin.sorted.idxstats")
                ).match("mirna_quant_bam") },

                { assert snapshot(
                    path("$outputDir/mirna_quant/mirtop/mirtop.tsv").exists(),
                    path("$outputDir/mirna_quant/mirtop/mirtop_rawData.tsv"),
                    path("$outputDir/mirna_quant/mirtop/mirtop.gff").exists(),
                    path("$outputDir/mirna_quant/mirtop/mirna.tsv"),
                    path("$outputDir/mirna_quant/mirtop/stats/mirtop_stats.txt").exists()
                ).match("mirna_quant_mirtop") },

                { assert snapshot(
                    path("$outputDir/mirna_quant/edger_qc/hairpin_counts.csv").exists(),
                    path("$outputDir/mirna_quant/edger_qc/hairpin_logtpm.csv").exists(),
                    path("$outputDir/mirna_quant/edger_qc/hairpin_unmapped_read_counts.txt").exists(),
                    path("$outputDir/mirna_quant/edger_qc/hairpin_logtpm.txt").exists(),
                    path("$outputDir/mirna_quant/edger_qc/hairpin_normalized_CPM.txt").exists(),
                    path("$outputDir/mirna_quant/edger_qc/mature_logtpm.csv").exists(),
                    path("$outputDir/mirna_quant/edger_qc/mature_counts.csv").exists(),
                    path("$outputDir/mirna_quant/edger_qc/mature_logtpm.txt").exists(),
                    path("$outputDir/mirna_quant/edger_qc/mature_unmapped_read_counts.txt").exists(),
                    path("$outputDir/mirna_quant/edger_qc/mature_normalized_CPM.txt").exists()
                ).match("mirna_quant_edger_qc") },

                { assert snapshot(
                    path("$outputDir/fastqc/raw/Clone1_N1.raw_fastqc.zip").exists(),
                    path("$outputDir/fastqc/raw/Clone1_N3.raw_fastqc.zip").exists()
                ).match("fastqc_raw") },

                { assert snapshot(
                    path("$outputDir/mirtrace/mirtrace-stats-contamination_basic.tsv"),
                    path("$outputDir/mirtrace/mirtrace-stats-rnatype.tsv"),
                    path("$outputDir/mirtrace/mirtrace-stats-phred.tsv"),
                    path("$outputDir/mirtrace/mirtrace-stats-contamination_detailed.tsv"),
                    path("$outputDir/mirtrace/mirtrace-stats-length.tsv"),
                    path("$outputDir/mirtrace/mirtrace-stats-qcstatus.tsv"),
                    path("$outputDir/mirtrace/mirtrace-stats-mirna-complexity.tsv")
                ).match("mirtrace") },

                { assert snapshot(
                    path("$outputDir/mirtrace/qc_passed_reads.all.collapsed/small_Clone1_N3.fastp.fasta"),
                    path("$outputDir/mirtrace/qc_passed_reads.all.collapsed/small_Clone1_N1.fastp.fasta")
                ).match("mirtrace_qc_passed_reads.all.collapsed") },

                { assert snapshot(
                    path("$outputDir/mirtrace/qc_passed_reads.rnatype_unknown.collapsed/small_Clone1_N3.fastp.fasta"),
                    path("$outputDir/mirtrace/qc_passed_reads.rnatype_unknown.collapsed/small_Clone1_N1.fastp.fasta")
                ).match("mirtrace_qc_passed_reads.rnatype_unknown.collapsed") },

                { assert snapshot(
                    path("$outputDir/bowtie_index/mirna_hairpin/fasta_bidx.3.ebwt"),
                    path("$outputDir/bowtie_index/mirna_hairpin/fasta_bidx.2.ebwt"),
                    path("$outputDir/bowtie_index/mirna_hairpin/fasta_bidx.1.ebwt"),
                    path("$outputDir/bowtie_index/mirna_hairpin/fasta_bidx.rev.2.ebwt"),
                    path("$outputDir/bowtie_index/mirna_hairpin/fasta_bidx.4.ebwt"),
                    path("$outputDir/bowtie_index/mirna_hairpin/fasta_bidx.rev.1.ebwt")
                ).match("bowtie_index_mirna_hairpin") },

                { assert snapshot(
                    path("$outputDir/bowtie_index/mirna_mature/fasta_bidx.2.ebwt"),
                    path("$outputDir/bowtie_index/mirna_mature/fasta_bidx.1.ebwt"),
                    path("$outputDir/bowtie_index/mirna_mature/fasta_bidx.4.ebwt"),
                    path("$outputDir/bowtie_index/mirna_mature/fasta_bidx.3.ebwt"),
                    path("$outputDir/bowtie_index/mirna_mature/fasta_bidx.rev.2.ebwt"),
                    path("$outputDir/bowtie_index/mirna_mature/fasta_bidx.rev.1.ebwt")
                ).match("bowtie_index_mirna_mature") },

                { assert snapshot(
                    path("$outputDir/multiqc/multiqc_data/fastqc-status-check-heatmap.txt"),
                    path("$outputDir/multiqc/multiqc_data/fastqc_overrepresented_sequences_plot.txt"),
                    path("$outputDir/multiqc/multiqc_data/mirtrace_complexity_plot.txt"),
                    path("$outputDir/multiqc/multiqc_data/multiqc_citations.txt"),
                    path("$outputDir/multiqc/multiqc_data/samtools-stats-dp.txt").exists(),
                    path("$outputDir/multiqc/multiqc_data/fastqc_sequence_length_distribution_plot.txt"),
                    path("$outputDir/multiqc/multiqc_data/multiqc_general_stats.txt").exists(),
                    path("$outputDir/multiqc/multiqc_data/fastqc_per_base_n_content_plot.txt"),
                    path("$outputDir/multiqc/multiqc_data/fastqc_per_sequence_quality_scores_plot.txt"),
                    path("$outputDir/multiqc/multiqc_data/mirtrace_qc_plot.txt"),
                    path("$outputDir/multiqc/multiqc_data/mirtrace_length_plot.txt"),
                    path("$outputDir/multiqc/multiqc_data/fastqc_top_overrepresented_sequences_table.txt").exists(),
                    path("$outputDir/multiqc/multiqc_data/fastqc_sequence_counts_plot.txt"),
                    path("$outputDir/multiqc/multiqc_data/mirtrace_rna_categories_plot.txt"),
                    path("$outputDir/multiqc/multiqc_data/samtools_alignment_plot.txt").exists(),
                    path("$outputDir/multiqc/multiqc_data/fastqc_per_base_sequence_quality_plot.txt"),
                    path("$outputDir/multiqc/multiqc_data/mirtop_read_count_plot.txt").exists(),
                    path("$outputDir/multiqc/multiqc_data/mirtop_unique_read_count_plot.txt").exists(),
                    path("$outputDir/multiqc/multiqc_data/mirtop_mean_read_count_plot.txt").exists(),
                    path("$outputDir/multiqc/multiqc_data/fastqc_sequence_duplication_levels_plot.txt"),
                    path("$outputDir/multiqc/multiqc_data/fastqc_per_sequence_gc_content_plot_Percentages.txt"),
                    path("$outputDir/multiqc/multiqc_data/fastqc_per_sequence_gc_content_plot_Counts.txt"),
                    path("$outputDir/multiqc/multiqc_data/multiqc_sources.txt").exists()
                ).match("multiqc_multiqc_data") },

            )
        }

    }

}
