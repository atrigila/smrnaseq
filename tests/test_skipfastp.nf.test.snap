{
    "mirtrace": {
        "content": [
            true,
            "mirtrace-stats-contamination_basic.tsv:md5,34005177b3038da76236b5d8fb468825",
            "mirtrace-stats-length.tsv:md5,847f38c5a6bbed4d3200ecaab05e090a",
            "mirtrace-stats-mirna-complexity.tsv:md5,d175bfcbcdebbdf768217da0078ad055",
            "mirtrace-stats-contamination_detailed.tsv:md5,141f6f46a4f9fcfac91303c125a2fc75",
            "mirtrace-stats-phred.tsv:md5,b55f5d6fe7a79dafd618e13f66505dfe",
            "mirtrace-stats-rnatype.tsv:md5,424cc10a33b239d01765f981dbe4b60c",
            "mirtrace-stats-qcstatus.tsv:md5,7c41c36e544b9aafe68bbeade3de9428",
            true,
            "mirtrace-stats-contamination_basic.tsv:md5,109ca0f96520aaa21bea6f55dc1b4b03",
            "mirtrace-stats-length.tsv:md5,e37044dc40483a893b577ebdbf4fa585",
            "mirtrace-stats-mirna-complexity.tsv:md5,bd8e19d2c40c3b1dbfec4037ca288b14",
            "mirtrace-stats-contamination_detailed.tsv:md5,d116d38bc104139f7496da66ce0a9436",
            "mirtrace-stats-phred.tsv:md5,3e1b63adcc5acd334059ebbaa5141b72",
            "mirtrace-stats-rnatype.tsv:md5,8a94743f0e853f8804ae225d4a588e6f",
            "mirtrace-stats-qcstatus.tsv:md5,469f837e7403e160eaec4c1cf3af0ba3"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-11T14:28:00.975156754"
    },
    "genome_quant_bam": {
        "content": [
            "Clone1_N3_mature_hairpin_genome.sorted.flagstat:md5,22d4dffd7b6fc1d2ae7827de3fb68ad7",
            "Clone1_N3_mature_hairpin_genome.sorted.idxstats:md5,2620288b88bba1ea3315414016c083a1",
            "Clone1_N1_mature_hairpin_genome.sorted.idxstats:md5,e0e4a95f5c21a926f7894cf1fbe3110b",
            "Clone1_N1_mature_hairpin_genome.sorted.flagstat:md5,62208acf0c7418d590b41318d7e17d67",
            "Clone1_N3_mature_hairpin_genome.sorted.stats:md5,91fdc857d4ed17552b2c3666e0bad342",
            "Clone1_N1_mature_hairpin_genome.sorted.stats:md5,ff101b4b88c7fbe2877b7e240e337bb6"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "23.10.0"
        },
        "timestamp": "2024-08-30T20:38:26.578558444"
    },
    "software_versions": {
        "content": [
            "{BOWTIE_MAP_GENOME={bowtie=1.3.1, samtools=1.14}, BOWTIE_MAP_HAIRPIN={bowtie=1.3.1, samtools=1.14}, BOWTIE_MAP_MATURE={bowtie=1.3.1, samtools=1.14}, BOWTIE_MAP_SEQCLUSTER={bowtie=1.3.1, samtools=1.14}, FASTQC_RAW={fastqc=0.12.1}, FORMAT_HAIRPIN={fastx_toolkit=0.0.14}, FORMAT_MATURE={fastx_toolkit=0.0.14}, INDEX_HAIRPIN={bowtie=1.3.1}, INDEX_MATURE={bowtie=1.3.1}, MIRTOP_QUANT={mirtop=0.4.25}, MIRTRACE_QC={mirtrace=1.0.1}, PARSE_HAIRPIN={seqkit=2.6.1}, PARSE_MATURE={seqkit=2.6.1}, SAMTOOLS_FLAGSTAT={samtools=1.2}, SAMTOOLS_IDXSTATS={samtools=1.2}, SAMTOOLS_INDEX={samtools=1.2}, SAMTOOLS_SORT={samtools=1.2}, SAMTOOLS_STATS={samtools=1.2}, SEQCLUSTER_COLLAPSE={seqcluster=1.2.9}, TABLE_MERGE={r-base=3.6.2}, Workflow={nf-core/smrnaseq=v2.3.2dev}}"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-10T13:31:01.195435949"
    },
    "mirna_quant_bam": {
        "content": [
            "Clone1_N3_mature_hairpin.sorted.flagstat:md5,1630edf055b591303d7c68d013745938",
            true,
            "Clone1_N3_mature_hairpin.sorted.stats:md5,4fbda712e789f98124583069f4f35187",
            true,
            "Clone1_N3_mature_hairpin.sorted.idxstats:md5,b44fb26f6be2accc7d52bc38efff69f4",
            true,
            "Clone1_N3_mature.sorted.stats:md5,e028b49e804c55056fca34090cd3d836",
            true,
            true,
            true,
            "Clone1_N3_mature.sorted.idxstats:md5,bde0293f0938a8a074ad3ac633d8cb73",
            "Clone1_N3_mature.sorted.flagstat:md5,9e287eb7ac83624b262864d0255217fd"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "23.10.0"
        },
        "timestamp": "2024-08-30T20:38:26.516593603"
    },
    "mirna_quant_edger_qc": {
        "content": [
            true,
            true,
            true,
            true,
            true,
            true,
            true,
            true,
            true,
            true
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "23.10.0"
        },
        "timestamp": "2024-08-30T20:38:26.540699935"
    },
    "multiqc_multiqc_data": {
        "content": [
            "fastqc-status-check-heatmap.txt:md5,a16737b9ae7b9b70b0ef7e462101a729",
            "fastqc_overrepresented_sequences_plot.txt:md5,0cacfefcbc37b292ae21564f845dfaa6",
            "mirtrace_complexity_plot.txt:md5,e27fb1e870985b3fe76744c027ce1c40",
            "multiqc_citations.txt:md5,f46d2983044658a4a89bdec5ba20fda3",
            "samtools-stats-dp.txt:md5,d1854b0ed73a4c9ae62a3a625c19d4b2",
            "fastqc_sequence_length_distribution_plot.txt:md5,ff2def0eab8321d4ed590b483641f43b",
            "multiqc_general_stats.txt:md5,4f9fbcd8b4395d3eb093aa242a484562",
            "fastqc_per_base_n_content_plot.txt:md5,c345fe5430e3a17ad1dbcc14e7595f50",
            "fastqc_per_sequence_quality_scores_plot.txt:md5,edf4d21e2928d37d94bb33a25e1d92a6",
            "mirtrace_qc_plot.txt:md5,75ee638eca3a02d61f2be321574488ed",
            "mirtrace_length_plot.txt:md5,7023ffcd95379998adbd65204b9998ee",
            true,
            "fastqc_sequence_counts_plot.txt:md5,da84941084250726e3828e8908640f8b",
            "mirtrace_rna_categories_plot.txt:md5,6f2b9465b699df7e5c7062b6495199af",
            "samtools_alignment_plot.txt:md5,2e7b7ce5aa4c69ee2b563c0376c40769",
            "fastqc_per_base_sequence_quality_plot.txt:md5,60f539c88c503680c0b2603749494948",
            "mirtop_read_count_plot.txt:md5,6820f657dc610579c53342b87fc2dcbb",
            "mirtop_unique_read_count_plot.txt:md5,5015a4e2f305a707b0d2e6ddc0498eb6",
            "mirtop_mean_read_count_plot.txt:md5,0148ef880eecae0c74b6a6ef2137e641",
            "fastqc_sequence_duplication_levels_plot.txt:md5,7e7eb4105b8f963bdf68e422e4ebce67",
            "fastqc_per_sequence_gc_content_plot_Percentages.txt:md5,7ac995de6a861676f64879b02d04f819",
            "fastqc_per_sequence_gc_content_plot_Counts.txt:md5,c18bf431a08ec1230720d83781e8903b",
            true
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-11T14:28:01.054069548"
    },
    "multiqc": {
        "content": [
            true
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "23.10.0"
        },
        "timestamp": "2024-08-30T20:38:26.610014871"
    },
    "mirna_quant_mirtop": {
        "content": [
            true,
            true,
            true,
            true,
            true
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "23.10.0"
        },
        "timestamp": "2024-08-30T20:38:26.512027897"
    }
}